### PR TITLE
Added AR module

### PR DESCRIPTION
**Pull request by SushrutHarness**
New Docs for AR module. To Do:
- Release notes
- KB?
- Docs landing page img: Light and dark
- Get Started guide (once done add links to landing page and home page)
- Content check to make sure open source version of docs line up with enterprise version
- LandingPage.tsx and featurelist (What does this do?)
- Future work: Roadmap. 

## Description

* Please describe your changes: \__________________________________
* Jira/GitHub Issue numbers (if any): \______________________________
* Preview links/images (Internal contributors only): \__________________

## PR lifecycle

We aim to merge PRs within one week or less, but delays happen sometimes.

If your PR is open longer than two weeks without any human activity, please tag a [code owner](https://github.com/harness/developer-hub/blob/main/.github/CODEOWNERS) in a comment.

PRs must meet these requirements to be merged:

- [ ] Successful preview build.
- [ ] Code owner review.
- [ ] No merge conflicts.
- [ ] Release notes/new features docs: Feature/version released to at least one prod environment.

### CODE REVIEW

File: src/components/Docs/ArtifactRegistry.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            </Link>
+          </div>
+        </div>
+        <div className={styles.spaceBetween}>
+          <div className={styles.content}>
+            <p>
+            Harness Artifact Registry (AR) is a universal hub for all your artifacts, regardless of type. By centralizing artifact management in one secure location, we're streamlining the process of handling multiple registries across various platforms and tools. 
+            </p>
+            <div className={styles.illustrationContainer}>
+              {/* <img
```
Comment (wei-harness) [inline src/components/Docs/ArtifactRegistry.tsx:42]: Do you still need an illustration for AR index page?
If yes, please create a ticket via https://form.asana.com/?k=kux3B0f4AoP6qzpyZSp0pQ&d=405956866894500

File: release-notes/artifact-registry.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+---
+title: Artifact Registry release notes
+sidebar_label: Artifact Registry
+date: 2024-09-26T20:00
+sidebar_position: 11
+---
+
```
Comment (wei-harness) [inline release-notes/artifact-registry.md:7]: Is it good to have an empty release-notes page?

File: sidebars.js [hunk 3]
```diff
@@ -498,6 +508,61 @@ const sidebars = {
...
+      customProps: {
+        description: "Learn about recent changes to Harness products.",
+      },
+    },
+     // Roadmap
+     {
+      type: "link",
+      label: "Roadmap",
+      className: "sidebar-roadmap",
+      href: "/roadmap/#ar",
```
Comment (wei-harness) [inline sidebars.js:540]: /roadmap/#ar is not a valid on road map page right now.

File: src/components/Docs/ArtifactRegistry.tsx (hunk 1 above)
Comment (SushrutHarness) [reply src/components/Docs/ArtifactRegistry.tsx:42]: CC @shankaranandh

**Review by pratmit — APPROVED** [review]
Approved, please get the module owner review before merging.

**Comment by SushrutHarness** [comment]
And @dewan-ahmed for the open source docs change.

**Review by wei-harness — APPROVED** [review]
Config files, ts(x) and css files look good to me.

**Review by rustd — APPROVED** [review]

**Pull request merged**